File: 0968-binary-tree-cameras/0968-binary-tree-cameras.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int cam = 0;
    int fun(TreeNode root){
        if(root == null) return 0;
        int l = fun(root.left);
        int r = fun(root.right);

        if(l == -1 || r == -1) //left have no cam or right has no camera
        {
            cam++;
            return 1; //install camera
        }
        if(l == 1 || r == 1) return 0; //left has cam or right has cam
        return -1; //no camera cover
    }
    public int minCameraCover(TreeNode root) {
        return fun(root)==-1?cam+1:cam;
    }
}

File: 0968-binary-tree-cameras/0968-binary-tree-cameras.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def __init__(self):
        self.cam = 0

    def fun(self, root):
        if root is None:
            return 0
        l = self.fun(root.left)
        r = self.fun(root.right)

        if l == -1 or r == -1:  # left has no cam or right has no camera
            self.cam += 1
            return 1  # install camera
        if l == 1 or r == 1:  # left has cam or right has cam
            return 0
        return -1  # no camera cover

    def minCameraCover(self, root):
        return self.fun(root) == -1 and self.cam + 1 or self.cam